vllm/entrypoints/chat_utils.py

```diff
@@ -368,6 +371,37 @@ def _parse_chat_message_content_parts(
     texts: List[str] = []
 
     mm_parser = mm_tracker.create_parser()
+    keep_multimodal_content = \
+        mm_tracker._model_config.hf_config.model_type in \
+            MODEL_KEEP_MULTI_MODAL_CONTENT
+
+    if keep_multimodal_content:
+        is_image = False
+        for part in parts:
+            part_type = part["type"]
+            if part_type == "text":
+                text = _TextParser(part)["text"]
+                texts.append(text)
+            elif part_type == "image_url":
+                image_url = _ImageParser(part)["image_url"]
+
+                if image_url.get("detail", "auto") != "auto":
+                    logger.warning(
+                        "'image_url.detail' is currently not supported and "
+                        "will be ignored.")
```

Comment:
IMO we should define a separate function for line 406 - 428 here, and reuse this function to deal with the special case when we need to keep multimodal content.